src/Java.Interop.Tools.TypeNameMappings/Java.Interop.Tools.TypeNameMappings/JavaNativeTypeManager.cs

```diff
@@ -535,6 +538,27 @@ public static string ToJniName (TypeDefinition type, IMetadataResolver? resolver
 		}
 
 		static string? ToJniNameFromAttributes (TypeDefinition type, IMetadataResolver? resolver)
+		{
+			return ToJniNameFromAttributesForAndroid (type, resolver) ??
+				ToJniNameFromAttributesForInterop (type, resolver);
+		}
+
+		static string? ToJniNameFromAttributesForInterop (TypeDefinition type, IMetadataResolver? resolver)
+		{
+			var attrs = type.CustomAttributes.Where (a =>
+				Resolve (resolver, a.AttributeType)
+				.FullName == "Java.Interop.JniTypeSignatureAttribute");
+			if (!attrs.Any ()) {
+				return null;
+			}
+			var attr    = attrs.Single ();
```

Comment:
Some of the LINQ here could be simplified to:
```suggestion
			var attr = type.CustomAttributes.FirstOrDefault (a =>
				Resolve (resolver, a.AttributeType)
				.FullName == "Java.Interop.JniTypeSignatureAttribute");
			if (attr == null) {
				return null;
			}
```

Reply:
Done.